### PR TITLE
Handle empty map case for Context.env

### PR DESCRIPTION
**Pull request by brienw**
Fixes [#101]
---

* [X] Documented the change if necessary
* [X] Tested this PRs change (with unit and/or functional tests)
* [X] Added this PRs change to CHANGELOG.md (in the `#master` section) if  necessary.

### CODE REVIEW

File: test/sshkit/context_test.exs [hunk 1]
```diff
@@ -42,5 +42,23 @@ defmodule SSHKit.ContextTest do
...
+    end
+
+    test "build with empty env" do
+      command =
+        @empty
+        |> SSHKit.env(%{})
+        |> Context.build("ls")
+
+      assert command == "/usr/bin/env ls"
+    end
```
Comment (pmeinhardt) [inline test/sshkit/context_test.exs]: I see we're using `SSHKit` in one of the tests above as well ("build with overwritten env"). I think we should avoid this when unit-testing `Context` and instead that test-case should go into `sshkit_test.exs`.

Does that make sense? If so, would you mind updating your PR to instead use the `%Context{@empty | env: …}` approach from the other tests?

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ https://github.com/bitcrowd/sshkit.ex/compare/v0.0.3...HEAD
 
 ### Fixes:
 
-* Put fixes here
+* Context properly handles the case where env is set to an empty map
```
Comment (pmeinhardt) [inline CHANGELOG.md]: We just merged a PR prep-ing the next release which added a bunch of "Fix" notes. A rebase onto our current master might be useful to verify this message ends up in the correct spot 😀 (maybe I'm underestimating git)

Comment (brienw) [reply CHANGELOG.md]: will do!

File: test/sshkit/context_test.exs (hunk 1 above)
Comment (pmeinhardt) [reply test/sshkit/context_test.exs]: Awesome! I'll can take care of moving the old test later on: https://github.com/bitcrowd/sshkit.ex/issues/103

File: CHANGELOG.md (hunk 2 above)
Comment (brienw) [reply CHANGELOG.md]: Shall i put these in the `0.1.0` Fix section of in the `master` Fix section?

Comment (pmeinhardt) [reply CHANGELOG.md]: You can put it in the `0.1.0` section 🤘

Comment (pmeinhardt) [reply CHANGELOG.md]: We haven't published that version yet, but are just preparing it.

**Comment by brienw** [comment]
Ok, there we go. I think i have everything in the right place now

**Pull request merged**